File: 프로그래머스/1/42862. 체육복/체육복.js
function solution(n, lost, reserve) {
    
    for (let i=0; i<reserve.length; i++) {
        if (lost.includes(reserve[i])) {
            lost.splice(reserve[i])
            reserve[i]=0
        }        
    }
    
    lost.sort()
    reserve.sort()
    
    for (let i=0; i<reserve.length; i++) {
        if (reserve[i]==0)
            continue
        else if (lost.includes(reserve[i]-1)) {
            lost.splice(reserve[i]-1)
        } else if (lost.includes(reserve[i]+1)) {
            lost.splice(reserve[i]+1)
        }
    
    if (lost.length>0)
        return n-lost.length
    else
        return n
    }
}

File: 프로그래머스/1/42862. 체육복/체육복.py
def solution(n, lost, reserve):
    lost.sort()
    reserve.sort()
    for num in reserve[:]:
        if num in lost:
            lost.remove(num)
            reserve.remove(num)
    for num in reserve:
        if num-1 in lost:
            lost.remove(num-1)
        elif num+1 in lost:
            lost.remove(num+1)
    if lost:
        return n-len(lost)
    else:
        return n